autospec/tarball.py

```diff
@@ -43,12 +41,107 @@
 tarball_prefix = ""
 gcov_file = ""
 archives = []
-go_archives = []
 giturl = ""
 domain = ""
 prefixes = dict()
 
 
+class Source(object):
+    """Holds data and methods for source code or archives management."""
+
+    def __init__(self, url, destination, path):
+        """Set default values for source file."""
+        self.url = url
+        self.destination = destination
+        self.path = path
+        self.type = None
+        self.prefix = None
+        self.subdir = None
+
+        # Extra  compressed archives
+        if not self.destination.startswith(':'):
+            self.get_type()
+            self.get_prefix()
+            self.extract()
+
+    def get_type(self):
+        """Determine compression type."""
+        if self.url.lower().endswith(('.zip', 'jar')):
+            self.type = 'zip'
+        elif self.url.lower().endswith('list'):
+            self.type = 'go'
+        else:
+            self.type = 'tar'
+
+    def get_prefix(self):
+        """Determine the prefix and subdir if no prefix."""
+        prefix_method = getattr(self, 'get_{}_prefix'.format(self.type))
+        prefix_method()
+        # When there is no prefix, create subdir
+        if not self.prefix:
+            self.subdir = os.path.splitext(os.path.basename(self.path))[0]
```

Comment:
I think I'd prefer the get_* to return a value. This is a set_*.

Reply:
Given that, would it make sense for this method to accept an optional value where the caller specifies the prefix to use?

Reply:
Or, maybe it should be find_*?